addons/medical_status/functions/fnc_addInventoryActions.sqf

```diff
@@ -0,0 +1,48 @@
+#include "script_component.hpp"
+/*
+ * Author: LinkIsGrim, johnb43
+ * Adds inventory and open backpack actions to uncon units.
+ *
+ * Arguments:
+ * 0: Unit <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [cursorTarget] call ace_medical_status_fnc_addInventoryActions
+ *
+ * Public: No
+ */
+
+if (!hasInterface) exitWith {};
+
+params ["_unit"];
+
+private _id = _unit addAction ["", {
+    params ["_target", "_caller"];
+
+    _caller action ["Gear", _target];
+}, nil, 5.1, true, true, "gear", toString {
+    (_target isNotEqualTo ACE_player) &&
+    {(lifeState _target) isEqualTo "INCAPACITATED"}
+}, 2];
+
+_unit setUserActionText [_id, localize "STR_ACTION_GEAR", "<img image='\A3\ui_f\data\igui\cfg\actions\gear_ca.paa' size='2.5' shadow=2 />"];
+
+_unit addAction ["OpenBag", {
+    params ["_target", "_caller"];
+
+    _caller action ["OpenBag", _target];
+}, nil, 5.2, true, true, "", toString {
+    private _backpackContainer = backpackContainer _target;
+    private _backpackConfig = configOf _backpackContainer;
+
+    (_target isNotEqualTo ACE_player) &&
+    {!((lifeState _target) in ["HEALTHY", "INJURED", "INCAPACITATED"])} &&
+    {!isNull _backpackContainer} &&
+    {!lockedInventory _backpackContainer} &&
+    {maxLoad _backpackContainer > 0} &&
+    {getNumber (_backpackConfig >> "disableInventory") != 1} &&
+    {_target setUserActionText [_actionId, format [localize "STR_ACTION_OPEN_BAG", getText (_backpackConfig >> "displayName")]]; true}
+}, 2];
```

Comment:
```suggestion
// Don't add actions twice
if (!isNil {_unit getVariable QGVAR(actions)}) exitWith {};

private _gearID = _unit addAction ["", {
    params ["_target", "_caller"];

    _caller action ["Gear", _target];
}, nil, 5.1, true, true, "gear", toString {
    (_originalTarget isNotEqualTo _this) &&
    {(lifeState _originalTarget) isEqualTo "INCAPACITATED"}
}, 2];

_unit setUserActionText [_gearID, localize "STR_ACTION_GEAR", "<img image='\A3\ui_f\data\igui\cfg\actions\gear_ca.paa' size='2.5' shadow=2 />"];

private _openBagID = _unit addAction ["OpenBag", {
    params ["_target", "_caller"];

    _caller action ["OpenBag", _target];
}, nil, 5.2, true, true, "", toString {
    private _backpackContainer = backpackContainer _originalTarget;
    private _backpackConfig = configOf _backpackContainer;

    (_originalTarget isNotEqualTo _this) &&
    {!((lifeState _originalTarget) in ["HEALTHY", "INJURED", "INCAPACITATED"])} &&
    {!isNull _backpackContainer} &&
    {!lockedInventory _backpackContainer} &&
    {maxLoad _backpackContainer > 0} &&
    {getNumber (_backpackConfig >> "disableInventory") != 1} &&
    {_originalTarget setUserActionText [_actionId, format [localize "STR_ACTION_OPEN_BAG", getText (_backpackConfig >> "displayName")]]; true}
}, 2];

_unit setVariable [QGVAR(actions), [_gearID, _openBagID]];
```

Figured it might be safer to switch over to `addAction`'s arguments whenever possible. However, if not wanted, the most important change here is the saving of actions' IDs. Technically they don't have to be saved, but I have a mod that removes scroll wheel actions, meaning if I can easily get the actions' IDs, I would be happy.

Reply:
setVars persist through respawn

Reply:
> setVars persist through respawn

Are you sure about this?

I know I have had to use the respawn EH to set ACE unit trait (such as doctor, engineer, etc.) on respawned units in the past, but it might have been another mod resetting all variables manually.

Regardless, in my testing, the solution I propose worked upon respawn, no issues whatsoever, which makes me believe that `setVariable` does not persist through respawn.